connection.go

```diff
@@ -124,12 +132,26 @@ func (c *Connection) Serve() {
 		c.Close()
 	}()
 
+	remoteAddr := trimStringFromSym(c.Conn.RemoteAddr().String())
+	strs := strings.Split(remoteAddr, ".")
+	remoteAddr = strs[0] + "." + strs[1]
+
 	if tlsConn, ok := c.Conn.(*tls.Conn); ok {
 		if err := tlsConn.Handshake(); err != nil {
+			c.Server.errorsMutex.Lock()
```

Comment:
another approach is to use channels here like (pseudo code):

```go
goodBadCh := nil

if ...; err != nil {
    ...
   goodBadCh <- {false, remoteAddr}
   return
}
   goodBadCh <- {true, remoteAddr}
   ...
```

and within a goroutine:
```go
   if bad {
       // increment counters
   } else {
       // increment other counters
   }
```

that would allow:
- to avoid logic complication of this functions and handle that explicitly
- to avoid usage of mutexes within a successful main flow but move it out